CONTRIBUTING.md

```diff
@@ -0,0 +1,137 @@
+
+| Milestone                     | Description                                                                                                                                                                       |
+|-------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Active Questions / Discussion | Refers to items where someone is actively investigating or answering a question. Items in this milestone should always have an owner.                                             |
+| Backlog                       | Refers to items that may need additional design or are expected to be delivered outside of the next 6 months                                                                      |
+| Sprint _X_                    | Refers to a specific work sprint during which this should be resolved                                                                                                             |
+| Triaged                       | Refers to items for which design is mostly understood and for which delivery should be within a 6 month period. Items in this milestone may just be waiting on sprint assignment. |
+| Unknown                       | Refers to any item for which no guarantees can be made. This is often a place for discussions and design.                                                                         |
+
+## Submitting a pull request
+
+Azure Functions welcomes pull requests. However, we ask that if you wish to address an issue tracked in one of our repositories that is not flagged with the `help-wanted` label, please check in with the maintainers to make sure work is not already in progress.
+
+Your pull request will now go through extensive checks by the project maintainers. Please be patient; the team operates a number of repositories and is often busy with their own contributions and service maintenance. We will try to get to your pull request as soon as we can. Please note that you may be asked to rebase or target a different branch as part of the review process, especially if other changes have been merged which would create a conflict.
+
+The general flow for making a change to an Azure Functions repository is:
+1. 🍴 Fork the repo (add the fork via `git remote add me <clone url here>`
+2. 🌳 Create a branch for your change (generally use dev) (`git checkout -b my-change`)
+3. 🛠 Make your change
+4. ✔️ Test your changes
+5. ⬆️ Push your changes to your fork (`git push me my-change`)
+6. 💌 Open a PR to the dev branch
+7. 📢 Address feedback and make sure tests pass (yes even if it's an "unrelated" test failure)
+8. 📦 [Squash](https://git-scm.com/docs/git-rebase) your changes into a meaningful commits (usually 1 commit) (`git rebase -i HEAD~N` where `N` is commits you want to squash)
```

Comment:
Is step 8 (squash) really necessary? Can't the project maintainer squash it in the GitHub UI when merging?

Reply:
We can make it optional or call out that a maintainer may do it. I think it's good to at least set expectations about what is going to happen throughout the process. Arguably we need a little more of that (I'm thinking of a quick runthrough of the label state transitions a PR might go through).

Reply:
My specific concerns around having contributors do their own squashing are:
1. It messes up the history, making it more difficult to review changes between iterations.
2. It makes it more likely that a contributor will make a mistake in the squash and lose their changes (I've seen people do this within our own team).

So if there isn't a specific reason we _need_ this, I would recommend leaving it out.